lib/libzfs/libzfs_sendrecv.c

```diff
@@ -1208,15 +1215,29 @@ send_progress_thread(void *arg)
 		tm = localtime(&t);
 		bytes = zc.zc_cookie;
 
-		if (pa->pa_parsable) {
-			(void) fprintf(stderr, "%02d:%02d:%02d\t%llu\t%s\n",
-			    tm->tm_hour, tm->tm_min, tm->tm_sec,
-			    bytes, zhp->zfs_name);
+		if (pa->pa_astitle) {
+			if (total > 0) {
+				pct = bytes / total;
+			} else
+				pct = 100;
+			if (pct > 100)
+				pct = 100;
+
+#ifdef HAVE_PROCTITLE
+			setproctitle("sending %s (%d%%: %llu/%llu)",
+			    zhp->zfs_name, pct, bytes, pa->pa_size);
+#endif /* HAVE_PROCTITLE */
 		} else {
-			zfs_nicebytes(bytes, buf, sizeof (buf));
-			(void) fprintf(stderr, "%02d:%02d:%02d   %5s   %s\n",
-			    tm->tm_hour, tm->tm_min, tm->tm_sec,
-			    buf, zhp->zfs_name);
+			if (pa->pa_parsable) {
+				(void) fprintf(stderr, "%02d:%02d:%02d\t%llu\t%s\n",
+				    tm->tm_hour, tm->tm_min, tm->tm_sec,
+				    bytes, zhp->zfs_name);
+			} else {
+				zfs_nicenum(bytes, buf, sizeof (buf));
```

Comment:
I believe you want zfs_nicebytes here instead.

Reply:
Ok :).